drivers/bluetooth/hci/Kconfig

```diff
@@ -238,3 +238,31 @@ config BT_SILABS_HCI_BUFFER_MEMORY
 	help
 	  Select the size of allocated memory buffer for the Silicon Labs
 	  Bluetooth Library.
+
+config BT_H4_NXP_CTLR
+	bool "NXP Bluetooth Controller"
```

Comment:
Wouldn't it make sense to add the following here:
```
default y
depends on DT_HAS_NXP_BT_HCI_UART_ENABLED
```

Then you can remove any explicit enablement in the app Kconfig overlays.

Reply:
...and then it'll also be easier for me to do the conversion as part of #72323

Reply:
> Wouldn't it make sense to add the following here:
> 
> ```
> default y
> depends on DT_HAS_NXP_BT_HCI_UART_ENABLED
> ```
> 
> Then you can remove any explicit enablement in the app Kconfig overlays.

Sorry, the dependency cannot be added.

The Bluetooth controller is module, it can work with any MCU with UART peripheral.

Reply:
@lylezhu2012 I'm not sure I follow. Even in the case that you describe, those MCUs that have this peripheral connected will still need to have the appropriate devicetree node with `compatible = "nxp,bt-hci-uart";`, won't they? If not, what is the purpose of the devicetree node?

Reply:
`compatible = "nxp,bt-hci-uart";` is the devicetree node required by NXP Bluetooth Controller. If you want to use NXP Bluetooth Controller, you need to add the devicetree node with `compatible = "nxp,bt-hci-uart";`.

Reply:
@lylezhu2012 the compatible strings are used to associate devicetree nodes with drivers. Which driver implements this compatible? You're adding the `hci_nxp_setup.c` extension driver in this same PR where you add the devicetree binding and compatible, so it seems natural to assume that these go together, even though the c-file currently doesn't `#define DT_DRV_COMPAT nxp_bt_hci_uart` (which I assumed was primarily because it only supports a single instance at the moment).

Reply:
Yes, `nxp,bt-hci-uart` is used for driver hci_nxp_setup.c.

> @lylezhu2012 the compatible strings are used to associate devicetree nodes with drivers. Which driver implements this compatible? You're adding the `hci_nxp_setup.c` extension driver in this same PR where you add the devicetree binding and compatible, so it seems natural to assume that these go together, even though the c-file currently doesn't `#define DT_DRV_COMPAT nxp_bt_hci_uart` (which I assumed was primarily because it only supports a single instance at the moment).

Yes. Currently, only one instance can be support. 

And I have a question, if does Bluetooth Host support multiple instance? I mean can we call `bt_enable` multiple times to enable multiple Bluetooth Host?

Reply:
> Yes, `nxp,bt-hci-uart` is used for driver hci_nxp_setup.c.

So in that case you should be able to apply my proposal wrt `default` and `depends on`. If there isn't a node with that compatible string it doesn't make sense to enable the driver.

> And I have a question, if does Bluetooth Host support multiple instance? I mean can we call `bt_enable` multiple times to enable multiple Bluetooth Host?

Not right now. This may become possible at some point in the future, especially thanks to #72323. With that PR you could quite easily construct a host that supports multiple controllers, or an app which exposes multiple controllers over different HCI transports.